setup.py

```diff
@@ -17,22 +50,72 @@
           'and is no longer supported by k2.')
     sys.exit(-1)
 
-# Refer to https://stackoverflow.com/questions/45150304/how-to-force-a-python-wheel-to-be-platform-specific-when-building-it
-# for why to introduce `bdist_wheel`.
-#
-# With `bdist_wheel`, the final wheel name looks like `k2-0.0.1.dev20201104-cp37-cp37m-linux_x86_64.whl`
 try:
     from wheel.bdist_wheel import bdist_wheel as _bdist_wheel
 
     class bdist_wheel(_bdist_wheel):
 
         def finalize_options(self):
             _bdist_wheel.finalize_options(self)
-            self.root_is_pure = False
+            if is_for_pypi():
+                # In this case, the generated wheel has a name in the form
+                # k2-xxx-pyxx-none-any.whl
+                self.root_is_pure = True
+            else:
+                # The generated wheel has a name ending with
+                # -linux_x86_64.whl
+                self.root_is_pure = False
 except ImportError:
     bdist_wheel = None
 
 
+def cmake_extension(name, *args, **kwargs) -> setuptools.Extension:
+    kwargs['language'] = 'c++'
+    sources = []
+    return setuptools.Extension(name, sources, *args, **kwargs)
+
+
+class BuildExtension(build_ext):
+
+    def build_extension(self, ext: setuptools.extension.Extension):
+        # build/temp.linux-x86_64-3.8
+        os.makedirs(self.build_temp, exist_ok=True)
+
+        # build/lib.linux-x86_64-3.8
+        os.makedirs(self.build_lib, exist_ok=True)
+
+        k2_dir = os.path.dirname(os.path.abspath(__file__))
+
+        cmake_args = os.environ.get('K2_CMAKE_ARGS', '')
+        make_args = os.environ.get('K2_MAKE_ARGS', '')
+
+        if cmake_args == '':
+            cmake_args = '-DCMAKE_BUILD_TYPE=Release'
```

Comment:
I think it was already suggested somewhere, but wouldn't `RelWithDebInfo` help solving issues reported by the users by giving more informative stack traces / gdb / cuda-gdb information?

Reply:
It was mentioned in https://github.com/k2-fsa/k2/pull/734#discussion_r629095496

> I am not sure whether it's worthwhile to change -O3 to -O2. If users want to report issues, I would
recommend them to use Debug.

I often see projects are released with `Release` build, not `RelWithDebInfo` build.

Reply:
Alright, missed that one. Cool, makes sense.